### ISSUE TITLE
1.28.1 release breaking google-cloud-php-spanner (Universe domain check for grpc transport)

### ISSUE THREAD

**Issue by matthewjumpsoffbuildings** [googleapis/gax-php#537]
I am using the google-cloud-php-spanner library, with grpc enabled etc. Everything was running fine on my local dev with the spanner emulator, until I recently ran a `composer update` and got the latest version of gax-php. Now I am getting an error

```
Typed property Google\ApiCore\CredentialsWrapper::$universeDomain must not be accessed before initialization
```

#### Environment details

  - OS: macOS 13.6
  - PHP version: 8.3
  - Package name and version: gax-php 1.28.1

#### Steps to reproduce

 Install the latest google-cloud-php-spanner library, along with  google/protobuf an grpc/grpc
 
 Run the spanner emulator, it wont have any instance

Attempt to use SpannerClient to see if an instance exists:

```php
$spanner = new SpannerClient(...);
$exists = $spanner->instance('instance_name')->exists());
```

Heres the relevant stacktrace

```
#0 /vendor/google/gax/src/Transport/GrpcTransport.php(260): Google\\ApiCore\\CredentialsWrapper->checkUniverseDomain()
#1 /vendor/google/gax/src/Transport/GrpcTransport.php(226): Google\\ApiCore\\Transport\\GrpcTransport->verifyUniverseDomain(Array)
#2 /vendor/google/gax/src/GapicClientTrait.php(631): Google\\ApiCore\\Transport\\GrpcTransport->startUnaryCall(Object(Google\\ApiCore\\Call), Array)
#3 /vendor/google/gax/src/Middleware/CredentialsWrapperMiddleware.php(58): Google\\Cloud\\Spanner\\Admin\\Instance\\V1\\Gapic\\InstanceAdminGapicClient->Google\\ApiCore\\{closure}(Object(Google\\ApiCore\\Call), Array)
#4 /vendor/google/gax/src/Middleware/FixedHeaderMiddleware.php(68): Google\\ApiCore\\Middleware\\CredentialsWrapperMiddleware->__invoke(Object(Google\\ApiCore\\Call), Array)
#5 /vendor/google/gax/src/Middleware/RetryMiddleware.php(89): Google\\ApiCore\\Middleware\\FixedHeaderMiddleware->__invoke(Object(Google\\ApiCore\\Call), Array)
#6 /vendor/google/gax/src/Middleware/OptionsFilterMiddleware.php(61): Google\\ApiCore\\Middleware\\RetryMiddleware->__invoke(Object(Google\\ApiCore\\Call), Array)
#7 /vendor/google/gax/src/GapicClientTrait.php(605): Google\\ApiCore\\Middleware\\OptionsFilterMiddleware->__invoke(Object(Google\\ApiCore\\Call), Array)
#8 /vendor/google/cloud-spanner/src/Admin/Instance/V1/Gapic/InstanceAdminGapicClient.php(937): Google\\Cloud\\Spanner\\Admin\\Instance\\V1\\Gapic\\InstanceAdminGapicClient->startCall('GetInstance', 'Google\\\\Cloud\\\\Sp...', Array, Object(Google\\Cloud\\Spanner\\Admin\\Instance\\V1\\GetInstanceRequest))
#9 [internal function]: Google\\Cloud\\Spanner\\Admin\\Instance\\V1\\Gapic\\InstanceAdminGapicClient->getInstance('projects/...', Array)
#10 /vendor/google/cloud-core/src/ExponentialBackoff.php(97): call_user_func_array(Array, Array)
#11 /vendor/google/cloud-core/src/GrpcRequestWrapper.php(135): Google\\Cloud\\Core\\ExponentialBackoff->execute(Array, Array)
#12 /vendor/google/cloud-core/src/GrpcTrait.php(81): Google\\Cloud\\Core\\GrpcRequestWrapper->send(Array, Array, Array)
#13 /vendor/google/cloud-spanner/src/Connection/Grpc.php(407): Google\\Cloud\\Spanner\\Connection\\Grpc->send(Array, Array)
#14 /vendor/google/cloud-spanner/src/Instance.php(279): Google\\Cloud\\Spanner\\Connection\\Grpc->getInstance(Array)
#15 /vendor/google/cloud-spanner/src/Instance.php(247): Google\\Cloud\\Spanner\\Instance->reload(Array)
#16 /app/Console/Commands/TestSpanner.php(26): Google\\Cloud\\Spanner\\Instance->exists()
```

If I pin my gax-php version to `1.27.2` in composer.json, the issue disappears

**Comment by acoulton** [comment]
Just to confirm this error applies to all usages of `InsecureCredentialsWrapper` from google/cloud-core - we're having the same issue connecting to a Cloud Tasks emulator.

I've proposed one possible solution at googleapis/google-cloud-php#7081 - my hunch is that's the right place to fix it, as that class is making quite a lot of assumptions about the internal / external API of CredentialsWrapper.

**Comment by surikman** [comment]
Same error here ✋ 
We are using `cloud-firestore-emulator` and this fails in our unit tests.

```
   ├ Error: Typed property Google\ApiCore\CredentialsWrapper::$universeDomain must not be accessed before initialization
   │
   │ /srv/www/vendor/google/gax/src/CredentialsWrapper.php:281
   │ /srv/www/vendor/google/gax/src/Transport/GrpcTransport.php:260
   │ /srv/www/vendor/google/gax/src/Transport/GrpcTransport.php:226
   │ /srv/www/vendor/google/gax/src/GapicClientTrait.php:631
   │ /srv/www/vendor/google/gax/src/Middleware/CredentialsWrapperMiddleware.php:58
   │ /srv/www/vendor/google/gax/src/Middleware/FixedHeaderMiddleware.php:68
   │ /srv/www/vendor/google/gax/src/Middleware/RetryMiddleware.php:89
   │ /srv/www/vendor/google/gax/src/Middleware/OptionsFilterMiddleware.php:61
   │ /srv/www/vendor/google/gax/src/Middleware/PagedMiddleware.php:66
   │ /srv/www/vendor/google/gax/src/GapicClientTrait.php:817
   │ /srv/www/vendor/google/gax/src/GapicClientTrait.php:774
   │ /srv/www/vendor/google/cloud-firestore/src/V1/Gapic/FirestoreGapicClient.php:847
   │ /srv/www/vendor/google/cloud-core/src/ExponentialBackoff.php:97
   │ /srv/www/vendor/google/cloud-core/src/GrpcRequestWrapper.php:135
   │ /srv/www/vendor/google/cloud-core/src/GrpcTrait.php:81
   │ /srv/www/vendor/google/cloud-firestore/src/Connection/Grpc.php:233
   │ /srv/www/vendor/google/cloud-core/src/Iterator/PageIteratorTrait.php:238
   │ /srv/www/vendor/google/cloud-core/src/Iterator/PageIteratorTrait.php:177
   │ /srv/www/vendor/google/cloud-core/src/Iterator/ItemIteratorTrait.php:136
```

**Comment by bshaffer** [comment]
Hello! Thank you for reporting this issue. 

This has been fixed in https://github.com/googleapis/google-cloud-php/pull/7077 and released in [google/cloud-core: 1.15.0](https://github.com/googleapis/google-cloud-php-core/releases/tag/v1.56.0). Please update your dependencies and you should be set. 

Sorry for the inconvenience.
